File: React-App/src/models/StudentExam.ts
import { StudenType } from "./Student"


export type StudentExamType = {
    id: number,
    teacherId: number | null,
    examId: number | null,
    examPath: string,
    namePrefix: string,
    isChecked: boolean,
    grade: number
    evaluation: string | null,
    checkedAt: Date | null,
    student:StudenType
}

File: Web API .NET/Server.Core/Entities/StudentExam.cs
using Server.Core.DTOs;
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace Server.Core.Entities
{
    public class StudentExam
    {
        public int Id { get; set; }

        public int ExamId { get; set; }
        public Exam Exam { get; set; }

        public int StudentId { get; set; }
        public User Student { get; set; }
        public string ExamPath { get; set; }
        public string StudentExamName { get; set; }
        public string ExamNamePrefix { get; set; }
        public int FolderId { get; set; }
        public Folder Folder { get; set; }
        public int TeacherId { get; set; }
        public User Teacher { get; set; }

        public bool IsChecked { get; set; }
        public int? Score { get; set; }
        public string? TeacherComments { get; set; }
        public DateTime? CheckedAt { get; set; }
    }
}
